docs/api.yaml

```diff
@@ -0,0 +1,641 @@
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /timetable/{id}:
+    get:
+      tags:
+        - "Timetable"
+      summary: "Find the full timetable of a student by his group"
+      parameters:
+        - name: "id"
+          in: "path"
+          description: "Identifier of the student to look at the timetable"
+          required: true
+          schema:
+            type: integer
+      operationId: "findTimetable"
+      responses:
+        200:
+          description: "List of courses with their timetables for a student"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/CoursePeriod'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /grades:
+    post:
+      tags:
+        - "Grade"
+      summary: "Create new grade of an enrolled course"
+      description: ""
+      operationId: "createGrade"
+      requestBody:
+        description: "The grade to attribute to the student and his teaching unit"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Grade'
+      responses:
+        200:
+          description: "The created grade"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Grade'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /interventions:
+    post:
+      tags:
+        - "Teacher"
+      summary: "Create new intervention hour of a teacher"
+      description: ""
+      operationId: "createIntervention"
+      requestBody:
+        description: "The interval of the intervention hours of the teacher"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Intervention'
+      responses:
+        200:
+          description: "The created intervention"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Intervention'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /group:
```

Comment:
No. PATCH the Student resource, and the PATCH-ed attribute is group. That is: PATCH /students/{id} with payload {'group_id' : 'uuid'}

Reply:
Is the group_id enough for this request ? 
Or is it the same as if I had all the groupe attribute (id and name) ?

Reply:
Only id used, fixed for all similars actions except for those who have specified other attributes, eg : main_teacher_name in course.

DELETE verb added to other endpoints. Please comment the 200 Response if needed before resolving this conversation.